setup-hakn/action.yaml

```diff
@@ -0,0 +1,178 @@
+    default: '{}'
+
+outputs:
+  load-balancer-ip:
+    description: |
+      The IP of the type LoadBalancer service on which knative serves.
+    value: ${{ steps.knative.outputs.load-balancer-ip }}
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Install Knative
+      id: knative
+      shell: bash
+      run: |
+        # Eliminates the resources blocks in a release yaml
+        function resource_blaster() {
+          local FILE="${1}"
+          local DOWNLOAD_URL="https://github.com/chainguard-dev/hakn/releases/download/v${{ inputs.version }}/${FILE}"
+
+          curl -L -s $DOWNLOAD_URL \
+            | yq e 'del(.spec.template.spec.containers[]?.resources)' - \
+            `# Filter out empty objects that come out as {} b/c kubectl barfs` \
+            | grep -v '^{}$'
+        }
+
+        function download_istioctl() {
+          ISTIO_VERSION=1.14.0
+          ISTIO_BASE_URL="https://github.com/istio/istio/releases/download"
+          case "$(echo $RUNNER_ARCH | awk '{print tolower($0)}')" in
+          x86|x64) ARCH=amd64;;
+          arm64)   ARCH=arm64;;
+          *)
+            echo Unsupported RUNNER_ARCH \"$RUNNER_ARCH\"
+            exit -1
+            ;;
+          esac
+          case "$(echo $RUNNER_OS | awk '{print tolower($0)}')" in
+          "linux") OS=linux;;
+          "macos") OS=osx;;
+          *)
+            echo Unsupported RUNNER_OS \"$RUNNER_OS\"
+            exit -1
+            ;;
+          esac
+          ISTIO_URL=${ISTIO_BASE_URL}/${ISTIO_VERSION}/istio-${ISTIO_VERSION}-${OS}-${ARCH}.tar.gz
+          wget $ISTIO_URL -O istio.tar.gz
+          tar xzf istio.tar.gz
+        }
+
+        download_istioctl
+        cat > istio-profile.yaml <<EOF
+        apiVersion: install.istio.io/v1alpha1
+        kind: IstioOperator
+        spec:
+          values:
+            global:
+              # Ideally for KinD installations we want to blow away the `resources`
+              # sections. However, doing that here will result in Istio's default which
+              # is high. Choosing some nomimal small numbers here:
+              defaultResources: &defaultResources
+                requests:
+                  cpu: 1m
+                  memory: 10Mi
+              proxy:
+                clusterDomain: ${{ inputs.cluster-domain }}
+                resources: *defaultResources
+
+          meshConfig:
+            defaultConfig:
+              terminationDrainDuration: "20s"
+
+          components:
+            ingressGateways:
+              - name: istio-ingressgateway
+                k8s: &defaultK8s
+                  hpaSpec:
+                    maxReplicas: 1
+                    minReplicas: 1
+                  resources: *defaultResources
+            pilot:
+              k8s: *defaultK8s
+        EOF
+        ./istio-${ISTIO_VERSION}/bin/istioctl install --skip-confirmation -f istio-profile.yaml
+
+        resource_blaster serving.yaml | kubectl apply --validate=false -f -
```

Comment:
nit: move function to where it is being used.

Reply:
Done